homeassistant/components/conversation/default_agent.py

```diff
@@ -118,25 +75,119 @@ async def async_process(
+
+            if not intents_dict:
+                _LOGGER.warning("No intents loaded")
+                return None
+
+            lang_intents = LanguageIntents(
+                intents=Intents.from_dict(intents_dict),
+                intents_dict=intents_dict,
+                loaded_components=set(self.hass.config.components),
+            )
+            self._lang_intents[language] = lang_intents
+        else:
+            # Check if any new components have been loaded
+            components_change = False
+            for component in self.hass.config.components:
+                if component not in lang_intents.loaded_components:
+                    # Check for sentences in this component with the target language
+                    sentences_dir = await self.async_get_sentences_dir(component)
+                    yaml_path = sentences_dir / f"{language}.yaml"
+                    if yaml_path.exists():
+                        # Merge sentences into existing dictionary
+                        _LOGGER.info("Loading intents YAML file %s", yaml_path)
+                        with open(yaml_path, encoding="utf-8") as yaml_file:
+                            merge_dict(lang_intents.intents_dict, safe_load(yaml_file))
+
+                        # Will need to recreate graph
+                        lang_intents.loaded_components.add(component)
+                        components_change = True
+
+            if components_change:
+                # This can be made faster by not re-parsing existing sentences.
+                # But it will likely only be called once anyways, unless new
+                # components with sentences are often being loaded.
+                lang_intents.intents = Intents.from_dict(lang_intents.intents_dict)
+
+        return lang_intents.intents
+
+    async def async_get_sentences_dir(self, component: str) -> Path:
+        """Get the sentences directory for a component."""
+        domain = component.rpartition(".")[-1]
+        integration = await async_get_integration(self.hass, domain)
+        return integration.file_path / "sentences"
+
+    def _make_areas_list(self) -> TextSlotList:
+        """Create slot list mapping area names/aliases to area ids."""
+        registry = area_registry.async_get(self.hass)
+        areas = []
+        for entry in registry.async_list_areas():
+            areas.append((entry.name, entry.id))
+            if entry.aliases:
+                for alias in entry.aliases:
+                    areas.append((alias, entry.id))
+
+        return TextSlotList.from_tuples(areas)
+
+    def _make_names_list(self) -> TextSlotList:
+        """Create slot list mapping entity names/aliases to entity ids."""
+        states = self.hass.states.async_all()
+        registry = entity_registry.async_get(self.hass)
+        names = []
+        for state in states:
+            names.append((state.name, state.entity_id))
+            entry = registry.async_get(state.entity_id)
```

Comment:
If a state has an entity registry entry and that entry has an `entity_category`, we should skip it. It's a configuration entry.